src/serac/physics/nonlinear_solid.cpp

```diff
@@ -77,27 +77,27 @@ NonlinearSolid::NonlinearSolid(std::shared_ptr<mfem::ParMesh> mesh, const Nonlin
   auto dim = mesh->Dimension();
   if (options.initial_displacement) {
     auto deform = options.initial_displacement->constructVector(dim);
-    setDisplacement(deform);
+    setDisplacement(*deform);
   }
 
   if (options.initial_velocity) {
     auto velo = options.initial_velocity->constructVector(dim);
-    setVelocity(velo);
+    setVelocity(*velo);
   }
   setViscosity(std::make_unique<mfem::ConstantCoefficient>(options.viscosity));
 
   for (const auto& [name, bc] : options.boundary_conditions) {
     // FIXME: Better naming for boundary conditions?
     if (name.find("displacement") != std::string::npos) {
       if (bc.coef_opts.isVector()) {
-        auto disp_coef = std::make_shared<mfem::VectorFunctionCoefficient>(bc.coef_opts.constructVector(dim));
+        std::shared_ptr<mfem::VectorCoefficient> disp_coef(bc.coef_opts.constructVector(dim));
```

Comment:
I'm a little surprised that this works - `shared_ptr` doesn't have a constructor that takes a `T&` (https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr)

Reply:
With a trivial example `std::shared_ptr<T>` can't be constructed from a `T` - https://godbolt.org/z/98e59j

Reply:
This works because `constructVector` and `constructScalar` now return unique pointers since we only know the base class.

Reply:
Ah, I missed that.  Thanks!